sdk/storage/azure-storage-blob-batch/src/main/java/com/azure/storage/blob/batch/BlobBatch.java

```diff
@@ -0,0 +1,445 @@
+        }
+
+        Disposable disposable = Flux.fromStream(batchOperationQueue.stream())
+            .flatMap(batchOperation -> batchOperation)
+            .subscribe();
+
+        while (!disposable.isDisposed()) {
+            // Wait until the batch operation has processed in the pipeline.
+            // This is used as opposed to block as it won't trigger an exception if ran in a Reactor thread.
+        }
+
+        this.batchRequest.add(ByteBuffer
+            .wrap(String.format("--%s--NEWLINE", batchBoundary).getBytes(StandardCharsets.UTF_8)));
+
+        return Flux.fromIterable(batchRequest);
+    }
+
+    long getContentLength() {
+        long contentLength = 0;
+
+        for (ByteBuffer request : batchRequest) {
+            contentLength += request.remaining();
+        }
+
+        return contentLength;
+    }
+
+    String getContentType() {
+        return String.format("multipart/mixed; boundary=%s", batchBoundary);
+    }
+
+    BlobBatchOperationResponse<?> getBatchRequest(int contentId) {
+        return batchMapping.get(contentId);
+    }
+
+    /*
+     * Enum class that indicates which type of operation this batch is using.
+     */
+    private enum BlobBatchType {
+        DELETE,
+        SET_TIER
+    }
+
+    /*
+     * This performs a cleanup operation that would be handled when the request is sent through Netty or OkHttp.
+     * Additionally, it removes the "x-ms-version" header from the request as batch operation requests cannot have this
+     * and it adds the header "Content-Id" that allows the request to be mapped to the response.
+     */
+    private Mono<HttpResponse> cleanseHeaders(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+        // Remove the "x-ms-header" as it shouldn't be included in the batch operation request.
+        context.getHttpRequest().getHeaders().remove(Constants.HeaderConstants.VERSION);
+
+        // Remove any null headers (this is done in Netty and OkHttp normally).
+        Map<String, String> headers = context.getHttpRequest().getHeaders().toMap();
+        headers.entrySet().removeIf(header -> header.getValue() == null);
+
+        context.getHttpRequest().setHeaders(new HttpHeaders(headers));
+
+        // Add the "Content-Id" header which allows this request to be mapped to the response.
+        context.getHttpRequest().setHeader(CONTENT_ID, context.getData(BATCH_REQUEST_CONTENT_ID).get().toString());
+
+        // Set the request URL to the correct endpoint.
+        try {
+            UrlBuilder requestUrl = UrlBuilder.parse(context.getHttpRequest().getUrl());
+            requestUrl.setPath(context.getData(BATCH_REQUEST_URL_PATH).get().toString());
+            context.getHttpRequest().setUrl(requestUrl.toURL());
+        } catch (MalformedURLException ex) {
+            throw Exceptions.propagate(logger.logExceptionAsError(new IllegalStateException(ex)));
+        }
```

Comment:
This doesn't belong in a cleanseHeaders method. Can you either rename or refactor?

Reply:
Made a new private method name `setRequestUrl`